x/lockup/spec/05_keeper.md

```diff
@@ -0,0 +1,36 @@
+<!--
+order: 5
+-->
+
+# Keepers
+
+## Lock Keeper
+
+Lock Keeper provide utility functions to query user's locks.
+
+```go
+// Keeper is the interface for lockup module keeper
+type Keeper interface {
+	// Return full balance of the module
+	GetModuleBalance(sdk.Context) sdk.Coins
+	// Return locked balance of the module
+	GetModuleLockedAmount(sdk.Context) sdk.Coins
+
+	// Returns whole balance deposited by the user which is not withdrawn yet
+	GetAccountPoolBalance(sdk.Context, sdk.AccAddress) sdk.Coins
+	// Return a locked amount that can't be withdrawn
+	GetAccountLockedAmount(sdk.Context, sdk.AccAddress) sdk.Coins
+
+	// Returns the total number of tokens of an account whose unlock time is beyond timestamp
+	GetAccountLockedPastTime(sdk.AccAddress, timestamp time.Time) (sdk.Coins | []types.PeriodLock)
```

Comment:
Should we also have an iterator of all of an account's locks whose unlocktime is _before_ a passed in timestamp?  (That way they can pass in current time, to query all their unlockable coins)

Reply:
Added, it can be useful